proxy/utils.go

```diff
@@ -0,0 +1,40 @@
+package proxy
+
+import (
+	"fmt"
+	"net/url"
+	"path"
+)
+
+const (
+	systemPath   = "/system/functions"
+	functionPath = "/system/function"
+)
+
+func createNamespaceEndpoint(gateway, namespace string) (string, error) {
+	gatewayURL, err := url.Parse(gateway)
+	if err != nil {
+		return "", fmt.Errorf("invalid gateway URL: %s", err.Error())
+	}
+	gatewayURL.Path = path.Join(gatewayURL.Path, systemPath)
+	if len(namespace) > 0 {
+		q := gatewayURL.Query()
+		q.Set("namespace", namespace)
+		gatewayURL.RawQuery = q.Encode()
+	}
+	return gatewayURL.String(), nil
+}
+
+func createFunctionNamespaceEndpoint(gateway, functionName, namespace string) (string, error) {
+	gatewayURL, err := url.Parse(gateway)
+	if err != nil {
+		return "", fmt.Errorf("invalid gateway URL: %s", err.Error())
+	}
+	gatewayURL.Path = path.Join(gatewayURL.Path, functionPath, functionName)
```

Comment:
For function we are using `functionname.namespace` or query params ? or we are supporting both option?

Reply:
Hey @viveksyngh, it is only query param, not `.`.

Reply:
I think I'd prepare a generic name instead of `createFunctionNamespaceEndpoint`.

Reply:
Will probably change the names of the methods with `createSystemEndpoint`/`createFunctionEndpoint`. Hope that is brief enough.

Reply:
Sounds viable :+1:

Reply:
changed :+1: